### PR TITLE
Adding InternalAPIKeyTestCase

### PR DESCRIPTION
**Pull request by HiranyaKavishani**
### Purpose
This PR will add the test cases for InternalAPIKeyAuthentication 

### Issues

Fixes #
Tests for https://github.com/wso2/product-microgateway/pull/1747

### Automation tests
 - Unit tests added: No
 - Integration tests added: Yes

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [ ] Validated respective github issues
- [ ] Assigned milestone to the github issue(s)

### CODE REVIEW

File: integration/test-integration/src/test/java/org/wso2am/micro/gw/tests/testCases/jwtValidator/InternalKeyTestCase.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    @Test(description = "Test to check the InternalKey is working")
+    public void invokeInternalKeyHeaderSuccessTest() throws Exception {
+        // Set header
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Internal-Key", internalKey);
+        HttpResponse response = HttpsClientRequest.doGet(Utils.getServiceURLHttps("/v2/pet/2") , headers);
+
+        Assert.assertNotNull(response);
+        Assert.assertEquals(response.getResponseCode(), HttpStatus.SC_OK,"Response code mismatched");
```
Comment (Rajith90) [inline integration/test-integration/src/test/java/org/wso2am/micro/gw/tests/testCases/jwtValidator/InternalKeyTestCase.java:48]: shall we validate the response message

File: integration/test-integration/src/test/java/org/wso2am/micro/gw/tests/testCases/jwtValidator/InternalKeyTestCase.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    @Test(description = "Test to check the internal key auth validate invalid signature key")
+    public void invokeInternalKeyHeaderInvalidTokenTest() throws Exception {
+        // Set header
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Internal-Key", TestConstant.INVALID_JWT_TOKEN);
+        HttpResponse response = HttpsClientRequest.doGet(Utils.getServiceURLHttps("/v2/pet/2") , headers);
+
+        Assert.assertNotNull(response);
+        Assert.assertEquals(response.getResponseCode(), HttpStatus.SC_UNAUTHORIZED,"Response code mismatched");
```
Comment (Rajith90) [inline integration/test-integration/src/test/java/org/wso2am/micro/gw/tests/testCases/jwtValidator/InternalKeyTestCase.java:59]: shall we validate the response message as well, whether it has 900901 code

File: integration/test-integration/src/test/java/org/wso2am/micro/gw/tests/testCases/jwtValidator/InternalKeyTestCase.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    @Test(description = "Test to check the internal key auth validate expired token")
+    public void invokeExpiredInternalKeyTest() throws Exception {
+
+        // Set header
+        Map<String, String> headers = new HashMap<String, String>();
+        headers.put("Internal-Key", TestConstant.EXPIRED_INTERNAL_KEY_TOKEN);
+        HttpResponse response = HttpsClientRequest.doGet(Utils.getServiceURLHttps("/v2/pet/2") , headers);
+
+        Assert.assertNotNull(response);
+        Assert.assertEquals(response.getResponseCode(), HttpStatus.SC_UNAUTHORIZED, "Response code mismatched");
```
Comment (Rajith90) [inline integration/test-integration/src/test/java/org/wso2am/micro/gw/tests/testCases/jwtValidator/InternalKeyTestCase.java:71]: shall we validate the response message as well, whether it has 900901 code

File: integration/test-integration/src/test/java/org/wso2am/micro/gw/tests/testCases/jwtValidator/InternalKeyTestCase.java (hunk 1 above)
Comment (HiranyaKavishani) [reply integration/test-integration/src/test/java/org/wso2am/micro/gw/tests/testCases/jwtValidator/InternalKeyTestCase.java:49]: Fixed by https://github.com/wso2/product-microgateway/pull/1758/commits/f891a8d6a045250156b9e6c4d2972f926d18e351

File: integration/test-integration/src/test/java/org/wso2am/micro/gw/tests/testCases/jwtValidator/InternalKeyTestCase.java (hunk 2 above)
Comment (HiranyaKavishani) [reply integration/test-integration/src/test/java/org/wso2am/micro/gw/tests/testCases/jwtValidator/InternalKeyTestCase.java:60]: Fixed by https://github.com/wso2/product-microgateway/pull/1758/commits/f891a8d6a045250156b9e6c4d2972f926d18e351

File: integration/test-integration/src/test/java/org/wso2am/micro/gw/tests/testCases/jwtValidator/InternalKeyTestCase.java (hunk 3 above)
Comment (HiranyaKavishani) [reply integration/test-integration/src/test/java/org/wso2am/micro/gw/tests/testCases/jwtValidator/InternalKeyTestCase.java:73]: Fixed by https://github.com/wso2/product-microgateway/pull/1758/commits/f891a8d6a045250156b9e6c4d2972f926d18e351

**Comment by HiranyaKavishani** [comment]
close the PR and open new https://github.com/wso2/product-microgateway/pull/1788 PR

**Pull request closed without merging**